### DISCUSSION TITLE
Access query data in local resolver without infinite recursion

### DISCUSSION THREAD

**Discussion by sighmun** [urql-graphql/urql#3259] [Q&A]
I'm trying use Local Resolvers to modify the output of a query, but I need to combine the cached results for that query with external data.

If I do a `readQuery` in a local resolver for the same query as the resolver, it infinitely recurses as it re-calls the same resolver.

For example:

```
resolvers: {
    Query: {
      todos: (parent, variables, cache) => {
        const cachedData = cache.readQuery{ todos: Todo[] }>({ query: gql`
          query {
            todos {
               ...Todo
            }
          }
        `, variables });

        ... // code that fetches otherData

        return cachedData.concat(otherData)
      }
    }
  }
```

Is there another way to access the cached data of that query that bypasses the local resolver? Or another way to achieve this?

Thanks in advance!

**Comment by kitten** [comment]
You likely want to use `cache.resolve` to read the value of a field: https://formidable.com/open-source/urql/docs/graphcache/local-resolvers/#resolving-other-fields

That gives you any value of the cached field. The TSDocs have more information on the exact API. In the case of this being a relation to other types it'll give you a key or a list of keys to other types, which `resolve` can accept again, as per the docs, but the cache itself understands how to continue reading from cached data even if you return relations as keys

**Comment by sighmun** [comment]
Thanks kitten! I'll give that a shot when I get a chance. I'd read about `cache.resolve` but thought it meant reconstructing links myself, I didn't realise this bit: `the cache itself understands how to continue reading from cached data even if you return relations as keys`
